artifact-version-diff/src/main/java/io/quarkus/qe/GenerateReport.java

```diff
@@ -0,0 +1,341 @@
+            fw.write(HTML_BASE_START);
+            fw.write(tableHeader);
+            for (String artifact : differentArtifacts.keySet()) {
+                List<String> versions = differentArtifacts.get(artifact).getDifferentVersions();
+                for (int i = 0; i < versions.size(); i++) {
+                    String writeColl = template;
+                    writeColl = i == 0 ? writeColl.replace("substitute-artifact", artifact) : writeColl.replace("substitute-artifact", "");
+                    writeColl = writeColl.replace("substitute-version", versions.get(i));
+                    writeColl = writeColl.replace("substitute-location", differentArtifacts.get(artifact).locationForVersion(versions.get(i)));
+                    fw.write(writeColl);
+                }
+            }
+            fw.write(HTML_BASE_END);
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to save output file. Log: " + e);
+        }
+    }
+
+    /**
+     * Generate list of files generated by versions-maven-plugin inside of Quarkus repository
+     * @return List of paths to generated files
+     */
+    public List<Path> generateListOfDiffFiles() {
+        String basePath = Objects.requireNonNull(System.getProperty("path-to-quarkus-repository"));
+        String versionDiffOutputFileName = Objects.requireNonNull(System.getProperty("version-plugin.report-output-file"));
+        Path basePathToQuarkusRepository = Path.of(basePath);
+
+        try (Stream<Path> entries = Files.walk(basePathToQuarkusRepository)) {
+            return entries.filter(Files::isRegularFile)
+                    .filter(f -> f.getFileName().toString().contains(versionDiffOutputFileName) &&
+                            (f.toAbsolutePath().toString().contains(basePathToQuarkusRepository + File.separator + "bom") ||
+                                    f.toAbsolutePath().toString().contains(basePathToQuarkusRepository + File.separator + "core") ||
+                                    f.toAbsolutePath().toString().contains(basePathToQuarkusRepository + File.separator + "extensions") ||
+                                    f.toAbsolutePath().toString().contains(basePathToQuarkusRepository + File.separator + "test-framework")))
+                    .toList();
+        } catch (IOException e) {
+            throw new RuntimeException("Error when generating list of diff files. Error log: " + e);
+        }
+    }
+
+    /**
+     * Comparing version if they are the same.
+     * Also modify them as they are differences between upstream and productized artifact
+     * @param upstreamVersion Version of upstream artifact
+     * @param downstreamVersion version of downstream artifact
+     * @return false if the version are not the same
+     */
+    public boolean versionComparator(String upstreamVersion, String downstreamVersion) {
+        String version = upstreamVersion.replace("-alpha", ".alpha");
```

Comment:
From testing it they change it 
Example with 3.8
`opentelemetry-semconv` in upstream is `1.21.0-alpha` but productized have it changed `1.21.0.alpha-redhat-00001`
Maybe they don't know about it but it's what happenning

Generaly what I saw it that they change `-` to `.` (which I miss here, found out yesterday)
Also if if upstream version have only major and minor they add patch version. Example can be `bcprov-jdk18on` which have upstream version `1.77` and productized version `1.77.0.redhat-00001`.
Probably standardized format is `major.minor.patch.<text>-redhat-NNNNN` and `major.minor.patch.redhat-NNNNN`

Reply:
Actually, you're right, partially, looking into it right now :D

The format is x.y.z(.QUALIFIER)*-redhat-NNNNN, so this assumption applies.

But also major.minor.patch.redhat-NNNNN doesn't match that, -redhat-NNNNN is what should be applied.

Anyway, just FYI, resolving otherwise.